src/sb_pointcloud_processing/include/BallExtractorNode.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Created By: Min Gyo Kim
+ * Created On: September 29, 2018
+ * Description: Header file for Ball Extractor Node
+ *  This node subscribes to a point cloud topic and expects the topic to contain
+ * a point cloud
+ *  that is filtered to contain roughly only the points that represent the
+ * tennis ball.
+ *  Then the node performs DBSCAN on the Y-Z plane, and calculates a rough
+ * estimate of the
+ *  center of the tennis ball, and publishes the point.
+ */
+
+#ifndef BALL_EXTRACTOR_IGVC_NODE_H
+#define BALL_EXTRACTOR_IGVC_NODE_H
+
+#include <DBSCAN.h>
+#include <RvizUtils.h>
+#include <iostream>
+#include <math.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <ros/console.h>
+#include <ros/ros.h>
+#include <sb_utils.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <string>
+
+class BallExtractorNode {
+  public:
+    /*
+     * @clusters stores the output from DBSCAN
+     */
+    std::vector<pcl::PointCloud<pcl::PointXYZ>> clusters;
```

Comment:
Member variables should be private and should be accessible only via getters. Here's a few reasons why this is important: https://stackoverflow.com/questions/1568091/why-use-getters-and-setters-accessors